tests/envs/test_action_dim_check.py

```diff
@@ -17,3 +33,59 @@ def test_serialize_deserialize(environment_id):
 
     with pytest.raises(ValueError, match="Action dimension mismatch"):
         env.step(0.1)
+
+
+@pytest.mark.parametrize("env", make_envs_by_action_space_type(spec_list, Discrete))
+def test_discrete_actions_out_of_bound(env):
+    # Car racing currently allow `out-of-bound` actions
```

Comment:
Why do we allow out of bounds actions for car racing?

Reply:
It seems to me that it checks only wether the action matches `steer`, `gas`, `brake` so if an out-of-bound action is performed the state of the car does not change.
https://github.com/openai/gym/blob/31e6f23e67a670e7276582710aa92534790bf92d/gym/envs/box2d/car_racing.py#L445-L454  
I think @jjshoots could give some hints, maybe I'm missing some behavior